server/graphql/v2/query/collection/ExpensesCollectionQuery.ts

```diff
@@ -396,12 +404,51 @@ export const ExpensesCollectionQueryResolver = async (
   } else if (args.tag === null || args.tags === null) {
     where['tags'] = { [Op.is]: null };
   }
-  if (args.minAmount) {
-    where['amount'] = { [Op.gte]: args.minAmount };
-  }
-  if (args.maxAmount) {
-    where['amount'] = { ...where['amount'], [Op.lte]: args.maxAmount };
+
+  if (args.amount?.gte || args.amount?.lte) {
+    if (args.amount.gte && args.amount.lte) {
+      assert(args.amount.gte.currency === args.amount.lte.currency, 'Amount range must have the same currency');
+    }
+    const currency = args.amount.gte?.currency || args.amount.lte?.currency;
+    const gte = args.amount.gte && getValueInCentsFromAmountInput(args.amount.gte);
+    const lte = args.amount.lte && getValueInCentsFromAmountInput(args.amount.lte);
+    const operator =
+      args.amount.gte && args.amount.lte
+        ? gte === lte
+          ? { [Op.eq]: gte }
+          : { [Op.between]: [gte, lte] }
+        : args.amount.gte
+          ? { [Op.gte]: gte }
+          : { [Op.lte]: lte };
+
+    where[Op.and].push(
+      sequelize.where(
+        sequelize.literal(`
+            CASE
+              WHEN "Expense"."data" #>> '{quote,sourceCurrency}' = '${currency}'
+                AND "Expense"."data" #>> '{quote,targetCurrency}' = "Expense"."currency"
+                THEN 1.0 / ("Expense"."data" #> '{quote,rate}')::NUMERIC
+              WHEN "Expense"."data" #>> '{quote,sourceCurrency}' = "Expense"."currency"
+                AND "Expense"."data" #>> '{quote,targetCurrency}' = '${currency}'
```

Comment:
Not a blocker since `currency` comes from an enum, but it's better to use query variables to prevent any risks of SQL injection.